rapid7_insightvm/komand_rapid7_insightvm/util/async_requests.py

```diff
@@ -0,0 +1,49 @@
+import aiohttp
+from typing import Collection
+from komand.exceptions import PluginException
+from .shared_resources import *
+
+
+class AsyncRequests:
+
+    def __init__(self, username, password):
+        self.auth = aiohttp.BasicAuth(login=username, password=password)
+
+    def get_async_session(self) -> aiohttp.ClientSession:
+        """
+        Create and return a new aiohttp ClientSession
+        :return: aiohttp ClientSession
+        """
+        return aiohttp.ClientSession(connector=aiohttp.TCPConnector(ssl=False), auth=self.auth)
+
+    @staticmethod
+    async def async_request(session, endpoint: str, method: str = 'get', params: Collection = None, payload: dict = None,
+                            json_response: bool = True):
+        """
+        Sends a asynchronous request to APIv3 with the provided endpoint and optional method/payload
+        :param session" The asynchronous session
+        :param endpoint: Endpoint for the APIv3 call defined in endpoints.py
+        :param method: HTTP method for the API request
+        :param params: URL parameters to append to the request
+        :param payload: JSON body for the API request if required
+        :param json_response: Boolean to return raw response
+        :return: Dict containing the JSON response body
+        """
+        if not params:
+            params = {}
+        if isinstance(params, list):
+            parameters = RequestParams.from_tuples(params)
+        else:
+            parameters = RequestParams.from_dict(params)
+        if payload:
+            response = await session.request(url=endpoint, method=method, json=payload, params=parameters.params)
+        else:
+            response = await session.request(url=endpoint, method=method, params=parameters.params)
```

Comment:
```suggestion
        extras = {"json": payload, "params": parameters.params}
        response = await session.request(url=endpoint, method=method, **extras)
```
I believe you can simplify this using the above code - the `**` will unpack the dictionary as possible into function arguments. You'd need to test and verify it works with the `request` call as it is dependent on default values in the function arguments.